aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -302,6 +300,36 @@ def set_max_seconds(self, max_wallclock_seconds):
         max_seconds = max_wallclock_seconds * max_seconds_factor
         self.ctx.inputs.parameters['CONTROL']['max_seconds'] = max_seconds
 
+    def set_restart_type(self, restart_type, parent_folder=None):
+        """Set the restart type for the next iteration."""
+
+        if parent_folder is None and restart_type != RestartType.FROM_SCRATCH:
+            raise ValueError('When not restarting from scratch, a `parent_folder` must be provided.')
```

Comment:
we are duplicating code from the `validate_inputs` but I think it is ok in this case. Just something to keep in mind if the validation logic changes. Maybe add a comment here that this is doubling checks from the validator

Reply:
Hmm, actually, this is not really duplicating code I think, since the `set_restart_type` method is not used to set the restart settings for the initial inputs, only for restarts via `process_handler`s. 

So it can be either considered superfluous or a good check on our use of the method when recovering from errors. ^^